### PR TITLE
Add Intel oneAPI env variables.

### PR DESCRIPTION
**Pull request by Neumann-A**
Required to use the Intel Fortran compiler instead of gfortran and probably also required to better find the mkl libraries?
Should I also add `I_MPI_ONEAPI_ROOT` (although that can be calculated by having ONEAPI_ROOT)?

### CODE REVIEW

File: src/vcpkg/base/system.process.cpp [hunk 1]
```diff
@@ -322,6 +322,9 @@ namespace vcpkg
             L"VULKAN_SDK",
             // Enable targeted Android NDK
             L"ANDROID_NDK_HOME",
+            // Environmental variables generated automatically by Intel oneAPI after installation
+            L"ONEAPI_ROOT",
+            L"IFORT_COMPILER19",
```
Comment (Neumann-A) [inline src/vcpkg/base/system.process.cpp:327]:
```suggestion
            L"IFORT_COMPILER19",
            L"IFORT_COMPILER20",
            L"IFORT_COMPILER21",
```

maybe move all those variables into the triplet via `VCPKG_ENV_PASSTHROUGH_UNTRACKED` ? 
And I don't mean only the ones I added I mean all of them.

**Review by BillyONeal — APPROVED** [review]

**Review by strega-nil-ms — APPROVED** [review]

**Comment by BillyONeal** [comment]
> Should I also add `I_MPI_ONEAPI_ROOT` (although that can be calculated by having ONEAPI_ROOT)?

I think that's more dependent on OneAPI rather than us

**Comment by Neumann-A** [comment]
> I think that's more dependent on OneAPI rather than us

I was thinking of more for what it could be useful. After checking `FindMPI` it is not using `I_MPI_ONEAPI_ROOT` but `I_MPI_ROOT`.  Changing the MPI implementation used within vcpkg probably requires a port overlay and a custom triplet anyway so I would say that it is not required to forward `I_MPI_ONEAPI_ROOT`

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**